### ISSUE TITLE
Ensure the data passed to the Audit Submission Access Page API is validated

### ISSUE THREAD

**Issue by geekygirlsarah** [GSA-TTS/FAC#334]
Breakout issue of #246, blocked until #246 is completed.

### Background

#246 is to add an API endpoint to save all the audit submission checklist steps. This turned out to be a much larger issue than originally anticipated due to it touching nearly all of the data models and the unrealized complexity of the data being saved. 

This ticket is to ensure the data being passed to the API endpoint goes through a more rigorous validation in order to complete #246 faster. This could involve adding/modifying validators/serializers attached to the `SingleAuditChecklist`, `User`, or `Access` models to ensure valid data and either accept/save or reject/don't save the data. 

### Acceptance Criteria (We'll know we're done when...)

- [ ] Invalid data passed to the API endpoint results in an error
- [ ] Error message(s) returned identify what failed on the submission
- [ ] Models have an easy way to identify invalid data passed into them

### Tasks

- [ ] TBD by person who takes the ticket or backend team

**Comment by petermarks-gsa** [comment]
Blocked by #246

**Comment by geekygirlsarah** [comment]
This issue is no longer needed as the evolution of #246 integrated better validation into it. I'm closing it.

**Comment by geekygirlsarah** [comment]
@petermarks-gsa I closed this as completed since the work involved here was done in #246.
